src/Cardano/ChainProducer/RustHttpBridge/Client.hs

```diff
@@ -34,3 +52,40 @@ getBlockByHash
     :<|> getTipBlockHeader
     = client api
 
+mkNetworkLayer :: Manager -> BaseUrl -> NetworkName -> NetworkLayer IO
+mkNetworkLayer mgr baseUrl network = NetworkLayer
+    { getBlock = \hash -> do
+            hash' <- hashToApi' hash
+            run (Api.getBlock <$> getBlockByHash network hash')
+    , getEpoch = \ep -> run (map Api.getBlock
+                             <$> getEpochById network (Api.EpochIndex ep))
+    , getNetworkTip = run (blockHeaderHash <$> getTipBlockHeader network)
+    }
+    where
+        run query = ExceptT $ (first convertError) <$> runClientM query env
+        env = mkClientEnv mgr baseUrl
+        convertError = NetworkLayerError . show
+
+newNetworkLayer :: Text -> IO (NetworkLayer IO)
+newNetworkLayer network = do
+    mgr <- newManager defaultManagerSettings
+    baseUrl <- parseBaseUrl "http://localhost:1337/"
+    pure $ mkNetworkLayer mgr baseUrl (Api.NetworkName network)
```

Comment:
Perhaps this should be moved into the `test` directory?

Reply:
Removed.